core/src/main/scala/pureconfig/package.scala

```diff
@@ -143,9 +145,9 @@ package object pureconfig {
    *
    * @param files Files ordered in decreasing priority containing part or all of a `Config`. Must not be empty.
    */
-  def loadConfigFromFiles[Config: ConfigConvert](files: Traversable[java.io.File]): Try[Config] = {
+  def loadConfigFromFiles[Config: ConfigConvert](files: Traversable[java.io.File]): ConfigReaderResult[Config] = {
     if (files.isEmpty) {
-      new scala.util.Failure(new IllegalArgumentException("The config files to load must not be empty."))
+      ConfigConvert.failWithThrowable[Config](new IllegalArgumentException("The config files to load must not be empty."))
```

Comment:
I think this can have a specialized `ConfigReaderFailure` instead of a `ThrowableFailure` with an `IllegalArgumentException`.